File: ListaDeLaCompra_copia/back/src/controller/tipos.controller.js
const db = require('../db/conection');
const getTipos = async (req, res) => {
  try {
    const [tipos, fields] = await db.query('SELECT DISTINCT tipo FROM ingredientes WHERE usuarioId IS NULL');
    if (tipos.length === 0) {
      return res.status(404).json({ message: 'No se encontraron tipos de ingredientes' });
    }
    res.json(tipos);
  } catch (error) {
    res.status(500).json({ error: 'Error interno del servidor' });
  }
}
module.exports = { getTipos };

File: ListaDeLaCompra_copia/back_py/controller/tipos_controller.py
from db.conexion import get_connection

def get_tipos():
    try:
        db = get_connection()
        cursor = db.cursor(dictionary=True)
        cursor.execute('SELECT DISTINCT tipo FROM ingredientes')
        tipos = cursor.fetchall()
        if not tipos:
            return {'message': 'No se encontraron tipos de ingredientes'}, 404
        return tipos, 200
    except Exception:
        return {'error': 'Error interno del servidor'}, 500
